src/components/swap/PendingModalContent/ProgressIndicator.tsx

```diff
@@ -0,0 +1,189 @@
+import { ChainId } from '@uniswap/sdk-core'
+import { useWeb3React } from '@web3-react/core'
+import { OrderContent } from 'components/AccountDrawer/MiniPortfolio/Activity/OffchainActivityModal'
+import Column from 'components/Column'
+import { Sign } from 'components/Icons/Sign'
+import { Swap } from 'components/Icons/Swap'
+import CurrencyLogo from 'components/Logo/CurrencyLogo'
+import Row from 'components/Row'
+import { SupportArticleURL } from 'constants/supportArticles'
+import { TransactionStatus } from 'graphql/data/__generated__/types-and-hooks'
+import { SwapResult } from 'hooks/useSwapCallback'
+import { UniswapXOrderStatus } from 'lib/hooks/orders/types'
+import useNativeCurrency from 'lib/hooks/useNativeCurrency'
+import { useEffect, useMemo, useState } from 'react'
+import { InterfaceTrade, TradeFillType } from 'state/routing/types'
+import { useOrder } from 'state/signatures/hooks'
+import { useIsTransactionConfirmed, useSwapTransactionStatus } from 'state/transactions/hooks'
+import styled from 'styled-components'
+import { Divider, ExternalLink } from 'theme/components'
+import { SignatureExpiredError } from 'utils/errors'
+
+import { ConfirmModalState } from '../ConfirmSwapModal'
+import { Step, StepDetails, StepStatus } from './Step'
+
+const StyledDivider = styled(Divider)`
+  margin: 16px 0px;
+`
+const StepConnector = styled.div`
+  width: 2px;
+  height: 24px;
+  background-color: ${({ theme }) => theme.neutral3};
+  margin: 4px 11px;
+`
+type ProgressIndicatorStep = Extract<
+  ConfirmModalState,
+  | ConfirmModalState.APPROVING_TOKEN
+  | ConfirmModalState.PERMITTING
+  | ConfirmModalState.PENDING_CONFIRMATION
+  | ConfirmModalState.WRAPPING
+  | ConfirmModalState.RESETTING_TOKEN_ALLOWANCE
+>
+export function ProgressIndicator({
+  steps,
+  currentStep,
+  trade,
+  swapResult,
+  wrapTxHash,
+  tokenApprovalPending = false,
+  revocationPending = false,
+  swapError,
+  inputTokenColor,
+  onRetryUniswapXSignature,
+}: {
+  steps: ProgressIndicatorStep[]
+  currentStep: ProgressIndicatorStep
+  trade?: InterfaceTrade
+  swapResult?: SwapResult
+  wrapTxHash?: string
+  tokenApprovalPending?: boolean
+  revocationPending?: boolean
+  swapError?: Error | string
+  inputTokenColor?: string
+  onRetryUniswapXSignature?: () => void
+}) {
+  const { chainId } = useWeb3React()
+  const nativeCurrency = useNativeCurrency(chainId)
+
+  const swapStatus = useSwapTransactionStatus(swapResult)
+  const order = useOrder(swapResult?.type === TradeFillType.UniswapX ? swapResult.response.orderHash : '')
```

Comment:
nit: rename `uniswapXOrder` to be more explicit

Reply:
Renamed